jena-arq/src/main/java/org/apache/jena/system/Txn.java

```diff
@@ -137,4 +139,46 @@ private static <T extends Transactional> void onThrowable(Throwable th, T txn) {
             txn.end();
         } catch (Throwable th2) { th.addSuppressed(th2); }
     }
+
+    /**
+     * Begins a transaction and returns a transaction control instance suitable
+     * for use with try-with-resources blocks.
+     * See {@link #begin(Transactional, TxnType)}.
+     */
+    public static AutoTxn begin(Transactional txn, ReadWrite readWrite) {
+        return begin(txn, TxnType.convert(readWrite));
+    }
+
+    /**
+     * Begins a transaction and returns a transaction control instance suitable
+     * for use with try-with-resources blocks.
+     * Allows for raising checked exceptions in an idiomatic way.
+     * Closing the TxnCtl instance will abort the transaction unless it
+     * has been manually committed.
+     * <p>
+     *
+     * Usage example:
+     * <pre>
+     * public void myMethod() throws IOException {
+     *   try (AutoTxn txn = Txn.begin(dataset, TxnType.WRITE)) {
```

Comment:
`begin` is rather overloaded. This "begin" isn't visibly paired with abort/commit in the application code.

`autoTxn` is a better name and works with static imports.

Reply:
Alternatively, the AutoTxn-related begin methods could be moved to the AutoTxn class.

`try (AutoTxn txn = AutoTxn.begin(transactional, TxnType.WRITE)) {`

Reply:
So in summary I tend towards to `Txn.autoTxn` solution.